assertion/function/analyzer.go

```diff
@@ -241,6 +257,175 @@ func run(pass *analysis.Pass) (result interface{}, _ error) {
 	return Result{FullTriggers: triggers, Errors: errs}, nil
 }
 
+// duplicateFullTriggersFromContractedFunctionsToCallers duplicates all the full triggers that have
+// FuncParam producer or UseAsReturn consumer or both, from the contracted functions to the callers
+// of all the contracted functions. This is necessary because we have created new
+// producers/consumers for argument pass or result return at every call site of the contracted
+// function. In order to connect such producers/consumers back to the contracted functions, we
+// create new full triggers that duplicates the original full triggers of the contracted functions
+// but uses the new producers/consumers instead.
+func duplicateFullTriggersFromContractedFunctionsToCallers(
+	pass *analysis.Pass,
+	funcContracts functioncontracts.Map,
+	funcTriggers [][]annotation.FullTrigger,
+	funcResults map[*types.Func]*functionResult,
+) {
+
+	// Find all the calls to contracted functions
+	// callsByCtrtFunc is a mapping contracted function -> caller -> call expression
```

Comment:
Just one question: Does this mean we support only one call-site per caller? What happens if a function calls a contracted function multiple times with different parameters? (That should probably be a test case too :) )

Reply:
No, it is totally OK to have multiple calls to a contracted functions with different parameters in a single function. Added a test case.